app/src/main/java/com/android/sample/ui/activity/CreateActivity.kt

```diff
@@ -256,15 +266,58 @@ fun CreateActivityScreen(
                 },
             )
             Spacer(modifier = Modifier.height(8.dp))
-            OutlinedTextField(
-                value = location,
-                onValueChange = { location = it },
-                label = { Text("Location") },
-                modifier = Modifier.padding(8.dp).fillMaxWidth().testTag("inputLocationCreate"),
-                placeholder = {
-                  Text(text = stringResource(id = R.string.request_location_activity))
-                },
-            )
+
+            // Location Input with dropdown using ExposedDropdownMenuBox
+            ExposedDropdownMenuBox(
+                expanded = showDropdown && locationSuggestions.isNotEmpty(),
+                onExpandedChange = { showDropdown = it }, // Toggle dropdown visibility ,
+            ) {
+              OutlinedTextField(
+                  value = locationQuery,
+                  onValueChange = {
+                    locationViewModel.setQuery(it)
+                    showDropdown = true // Show dropdown when user starts typing
+                  },
+                  label = { Text("Location") },
+                  placeholder = { Text("Enter an Address or Location") },
+                  modifier =
+                      Modifier.menuAnchor() // Anchor the dropdown to this text field
+                          .padding(8.dp)
+                          .fillMaxWidth()
+                          .testTag("inputLocationCreate"),
+                  singleLine = true)
+
+              // Dropdown menu for location suggestions
+              ExposedDropdownMenu(
+                  expanded = showDropdown && locationSuggestions.isNotEmpty(),
+                  onDismissRequest = { showDropdown = false }) {
+                    locationSuggestions.filterNotNull().take(3).forEach { location ->
+                      DropdownMenuItem(
+                          text = {
+                            Text(
+                                text =
+                                    location.name.take(30) +
+                                        if (location.name.length > 30) "..."
+                                        else "", // Limit name length
+                                maxLines = 1 // Ensure name doesn't overflow
+                                )
```

Comment:
Good job, Ui wise and logic wise to show suggestions of location and save the location choosed